src/foam/mlang/mlangJava.js

```diff
@@ -323,6 +323,17 @@ foam.CLASS({
 });
 
 
+foam.CLASS({
+  refines: 'foam.mlang.predicate.Has',
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: 'return getArg1().f(obj) != null && getArg1().f(obj) != "";'
```

Comment:
Testing for an empty array is not straight forward.  Testing for an array can be accomplished with 'value.getClass().isArray()' or 'instanceof _Type_[]', but then to get the length, it needs to be cast or you have to know the _Type_.